Phönix: Exception bei GeoLocations

Im Phönix-Crawler tritt bei der Verarbeitung der GeoLocations eine Exception auf, die das Beenden des Suchlaufs verhindert:

> Caused by: java.lang.IllegalArgumentException: No enum constant mServer.crawler.sender.newsearch.GeoLocations.de
	at java.lang.Enum.valueOf(Unknown Source)
	at mServer.crawler.sender.newsearch.GeoLocations.valueOf(GeoLocations.java:6)
	at mServer.crawler.sender.phoenix.parser.ZdfDownloadDtoDeserializer.parseGeoLocation(ZdfDownloadDtoDeserializer.java:104)
	at mServer.crawler.sender.phoenix.parser.ZdfDownloadDtoDeserializer.deserialize(ZdfDownloadDtoDeserializer.java:55)
	at mServer.crawler.sender.phoenix.parser.ZdfDownloadDtoDeserializer.deserialize(ZdfDownloadDtoDeserializer.java:21)
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
	at com.google.gson.Gson.fromJson(Gson.java:887)
	at com.google.gson.Gson.fromJson(Gson.java:852)
	at com.google.gson.Gson.fromJson(Gson.java:801)
	at mServer.crawler.sender.phoenix.tasks.ZdfTaskBase.deserializeOptional(ZdfTaskBase.java:40)
	at mServer.crawler.sender.phoenix.tasks.PhoenixFilmDetailTask.processRestTarget(PhoenixFilmDetailTask.java:85)
	at mServer.crawler.sender.phoenix.tasks.AbstractRestTask.processElement(AbstractRestTask.java:68)
	at mServer.crawler.sender.phoenix.tasks.AbstractRestTask.processElement(AbstractRestTask.java:29)
	at mServer.crawler.sender.orf.tasks.AbstractRecursivConverterTask.processElements(AbstractRecursivConverterTask.java:58)
	at mServer.crawler.sender.orf.tasks.AbstractRecursivConverterTask.compute(AbstractRecursivConverterTask.java:65)
	at mServer.crawler.sender.orf.tasks.AbstractRecursivConverterTask.compute(AbstractRecursivConverterTask.java:23)
	at java.util.concurrent.RecursiveTask.exec(Unknown Source)
	at java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.util.concurrent.ForkJoinPool.helpStealer(Unknown Source)
	at java.util.concurrent.ForkJoinPool.awaitJoin(Unknown Source)
	at java.util.concurrent.ForkJoinTask.doJoin(Unknown Source)
	at java.util.concurrent.ForkJoinTask.join(Unknown Source)
	at mServer.crawler.sender.orf.tasks.AbstractRecursivConverterTask.compute(AbstractRecursivConverterTask.java:72)
	at mServer.crawler.sender.orf.tasks.AbstractRecursivConverterTask.compute(AbstractRecursivConverterTask.java:71)

--- a/src/main/java/mServer/crawler/sender/phoenix/parser/ZdfDownloadDtoDeserializer.java
+++ b/src/main/java/mServer/crawler/sender/phoenix/parser/ZdfDownloadDtoDeserializer.java
@@ -100,12 +100,13 @@ public class ZdfDownloadDtoDeserializer implements JsonDeserializer<Optional<Dow
       final JsonElement geoLocation = attributes.getAsJsonObject().get(JSON_ELEMENT_GEOLOCATION);
       if (geoLocation != null) {
         final JsonElement geoValue = geoLocation.getAsJsonObject().get(JSON_PROPERTY_VALUE);
-        if (geoValue != null && !geoValue.getAsString().equals("none")) {
-          final Optional<GeoLocations> foundGeoLocation = Optional.of(GeoLocations.valueOf(geoValue.getAsString()));
-          if (foundGeoLocation.isPresent()) {
-            dto.setGeoLocation(foundGeoLocation.get());
-          } else {
-            LOG.debug(String.format("Can't find a GeoLocation for \"%s", geoValue.getAsString()));
+        if (geoValue != null && !geoValue.getAsString().toUpperCase().equals("NONE")) {
+          switch (geoValue.getAsString().toUpperCase()) {
+            case "DE":
+              dto.setGeoLocation(GeoLocations.GEO_DE);
+              break;
+            default:
+              LOG.debug(String.format("Can't find a GeoLocation for \"%s", geoValue.getAsString()));
           }
         }
       }

--- a/src/main/java/mServer/crawler/sender/phoenix/tasks/PhoenixFilmDetailTask.java
+++ b/src/main/java/mServer/crawler/sender/phoenix/tasks/PhoenixFilmDetailTask.java
@@ -123,6 +123,9 @@ public class PhoenixFilmDetailTask extends ZdfTaskBase<DatenFilm, CrawlerUrlDTO>
     if (downloadDto.getSubTitleUrl().isPresent()) {
       CrawlerTool.addUrlSubtitle(film, downloadDto.getSubTitleUrl().get());
     }
+    if (downloadDto.getGeoLocation().isPresent()) {
+      film.arr[DatenFilm.FILM_GEO] = downloadDto.getGeoLocation().get().getDescription();
+    }
     taskResults.add(film);
   }
 

